pyiron/atomistics/structure/atoms.py

```diff
@@ -1797,10 +1797,12 @@ def f_ind_ext(x):
             )
         self.min_nbr_number = min_nbr
         self.max_nbr_number = max_nbr
-        neighbor_obj.distances = self.neighbor_distance
-        neighbor_obj.vecs = self.neighbor_distance_vec
-        neighbor_obj.indices = self.neighbor_index
-        neighbor_obj.shells = self.neighbor_shellOrder
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always", np.VisibleDeprecationWarning)
```

Comment:
I'm not really sure if we're allowed to do this. This does not change the overall filter?

Reply:
No, it only captures the warnings for the indended code section under `with ..`. Some sample code below

```
import warnings

def raise_warning():
    warnings.warn("Sample warning")

with warnings.catch_warnings(record=True) as w:
    #warnings.simplefilter("always", UserWarning)
    for _ in range(3):
        raise_warning()
```

and then in a separate cell if you type `raise_warning()`, it should raise the warning. Isn't this okay?

Reply:
Wait I don't really understand why this example explains that the overall filter is not changed with `warnings.simplefilter` in the code above.

Reply:
Because calling `raise_warning` outside the with statement still raises the warning.

Reply:
That means we don't need `warnings.simplefilter` in the end ... ? Sorry I'm probably not getting the point.

Reply:
This does not change any functionality. It's just that with the latest `numpy` version, every time we create an array with elements of unequal sizes, numpy complains. This is unavoidable when using `get_neighbors` which results in a lot of constant annoying warning messages (see image below). These changes capture them!

![image](https://user-images.githubusercontent.com/17953127/92368159-a4c23200-f0f7-11ea-9097-cc2a098f3427.png)

Reply:
No wait Sudarsan, my concern is different: Doesn't setting `warnings.simplefilter` change the global filter? @jan-janssen and I think pyiron shouldn't change the filter that user might be using.

Reply:
For example, what happens if the user has a code like this:

```
import warnings

warnings.simplefilter('default') # filter changed to default

neighbours = structure.get_neighbors() # filter changed to always?
```

Reply:
> 
> 
> For example, what happens if the user has a code like this:
> 
> ```
> import warnings
> 
> warnings.simplefilter('default') # filter changed to default
> 
> neighbours = structure.get_neighbors() # filter changed to always?
> ```

The filter won't change. You can check by calling `print(warnings.filters)`. Anyway based on Jan's comment let's discuss this pull request during the pyiron meeting? It's not that critical anyway